ros2topic/ros2topic/api/__init__.py

```diff
@@ -164,38 +166,39 @@ def qos_profile_from_short_keys(
 ) -> rclpy.qos.QoSProfile:
     """Construct a QoSProfile given the name of a preset, and optional overrides."""
     # Build a QoS profile based on user-supplied arguments
+    if not preset_profile:
+        preset_profile = default_preset_str
     profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(preset_profile)
     if history:
         profile.history = rclpy.qos.QoSHistoryPolicy.get_from_short_key(history)
     if durability:
         profile.durability = rclpy.qos.QoSDurabilityPolicy.get_from_short_key(durability)
     if reliability:
         profile.reliability = rclpy.qos.QoSReliabilityPolicy.get_from_short_key(reliability)
-    if depth >= 0:
+    if depth and depth >= 0:
         profile.depth = depth
     else:
         if (profile.durability == rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL
                 and profile.depth == 0):
             profile.depth = 1
-
     return profile
 
 
 def add_qos_arguments_to_argument_parser(
-    parser: argparse.ArgumentParser, is_publisher: bool = True, default_preset: str = 'sensor_data'
+    parser: argparse.ArgumentParser, is_publisher: bool = True,
+    default_preset: str = default_preset_str
 ) -> None:
     """Extend an existing ArgumentParser to allow input of QoS policy overrides."""
     verb = 'publish' if is_publisher else 'subscribe'
     parser.add_argument(
         '--qos-profile',
         choices=rclpy.qos.QoSPresetProfiles.short_keys(),
-        default=default_preset,
         help='Quality of service preset profile to {} with (default: {})'
              .format(verb, default_preset))
     default_profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(
         default_preset)
```

Comment:
Technically, the default values reported for the `--qos-reliability` and `--qos-durability` options below are incorrect for `ros2 topic echo`. By default, we are automatically matching existing publishers. Maybe we should add separate parser options for `echo` instead of reusing them between verbs.

Reply:
Addressed on [160fce7](https://github.com/ros2/ros2cli/pull/613/commits/160fce7d4411ce53251d12ce92dd55b8e77fc046)